knimin/lib/data_access.py

```diff
@@ -2399,6 +2399,134 @@ def getProjectNames(self):
         sql = """SELECT project FROM project"""
         return [x[0] for x in self._con.execute_fetchall(sql)]
 
+    def get_plate_info(self, plate_id):
+        """Gets attributes of a plate by id
+
+        Parameters
+        ----------
+        plate_id : int
+            Plate id
+
+        Returns
+        -------
+        dict
+            Plate attributes
+        """
+        sql = """SELECT * FROM pm.plate WHERE plate_id = %s"""
+        return self._con.execute_fetchdict(sql, [plate_id])[0]
+
+    def set_plate_info(self, plate_id, plate_info):
```

Comment:
I think that what you're saying actually reinforces my point. The "create_plate" and "set_plate_info" are two different operations that need to execute different checks.

I know the database is doing those checks for you, but debugging with errors from SQL has been hard in the past.

This actually exposed the need for a better API. Instead of accepting a dictionary with all possible data, change the API to be verbose, and specifically ask for the parameters that are needed, while letting the other one nullable. An example of this idea is in the artifact creation in Qiita: https://github.com/biocore/qiita/blob/master/qiita_db/artifact.py#L159.

Then you can create the "set" function to change the parameters that make sense to change, not all of them are going to be 'setable'. For example, the user that creates the plate.

You can read that link that I sent above regarding the artifact object in Qiita. It will help you to understand how we use python to limit what can be done in the DB and provide useful errors.